python/cudf/cudf/core/column/categorical.py

```diff
@@ -1472,6 +1463,53 @@ def view(self, dtype: Dtype) -> ColumnBase:
             "Categorical column views are not currently supported"
         )
 
+    @staticmethod
+    def _concat(objs: MutableSequence[CategoricalColumn]) -> CategoricalColumn:
+        # TODO: This function currently assumes it is being called from
+        # column._concat_columns, at least to the extent that all the
+        # preprocessing in that function has already been done. That should be
+        # improved as the concatenation API is solidified.
+
+        # Find the first non-null column:
+        head = next((obj for obj in objs if obj.valid_count), objs[0])
+
+        # Combine and de-dupe the categories
+        cats = (
+            cudf.concat([o.cat().categories for o in objs])
+            .to_series()
+            .drop_duplicates(ignore_index=True)
```

Comment:
Should probably move `drop_duplicates` down to column or something like that instead of converting column --> Series --> column here.

Reply:
This is definitely worth doing, but I'm going to punt it down the road right now. Currently `drop_duplicates` is relying on a stream compaction algorithm from `libcudf` that's implemented on a `table` because we need to be able to measure uniqueness across multiple columns. I think the easiest path forward would probably be to implement a `Column` method that creates a new `table` with one `column` under the hood and operate on that, but there may be a better solution that will emerge as more of our Python code is consolidated and I think doing that properly is out of scope for this PR.

Reply:
Is there at least a way to do it with a generic Frame object instead of a Series? There's so much gunk in Series that any time we can get rid of it from our low level functions we should.

Reply:
Ah. Yes, this is just me getting tunnel-visioned and into the `Column` question. The `cudf.concat` call here is, for better or worse, returning an `Index` object, not a `Column`, so we can remove the `to_series` call here entirely and still have this work with no additional modifications.